cluster/get-kube.sh

```diff
@@ -63,7 +63,49 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+KUBERNETES_RELEASE_URL_OVERRIDDEN=false
+KUBERNETES_CI_RELEASE_URL_OVERRIDDEN=false
+
+if [[ -v KUBERNETES_RELEASE_URL ]]; then
+    KUBERNETES_RELEASE_URL_OVERRIDDEN=true
+fi
+if [[ -v KUBERNETES_CI_RELEASE_URL ]]; then
+    KUBERNETES_CI_RELEASE_URL_OVERRIDDEN=true
+fi
+
 KUBERNETES_RELEASE_URL="${KUBERNETES_RELEASE_URL:-https://storage.googleapis.com/kubernetes-release/release}"
+KUBERNETES_CI_RELEASE_URL="${KUBERNETES_CI_RELEASE_URL:-https://storage.googleapis.com/kubernetes-release-dev/ci}"
+
+if [[ ${KUBERNETES_RELEASE_URL_OVERRIDDEN} = true && ${KUBERNETES_CI_RELEASE_URL_OVERRIDDEN} = false ]]; then
+    KUBERNETES_CI_RELEASE_URL=${KUBERNETES_RELEASE_URL}
+fi
+
+KUBE_RELEASE_VERSION_REGEX="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)(-(beta|alpha)\\.(0|[1-9][0-9]*))?$"
+KUBE_CI_VERSION_REGEX="^v(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)\\.(0|[1-9][0-9]*)-(beta|alpha)\\.(0|[1-9][0-9]*)(\\.(0|[1-9][0-9]*)\\+[-0-9a-z]*)?$"
+
+# Sets KUBE_VERSION variable if an explicit version number was provided (e.g. "v1.0.6",
+# "v1.2.0-alpha.1.881+376438b69c7612") or resolves the "published" version
+# <path>/<version> (e.g. "release/stable",' "ci/latest-1") by reading from GCS.
+#
+# See the docs on getting builds for more information about version
+# publication.
+#
+# Args:
+#   $1 version string from command line
+# Vars set:
+#   KUBE_VERSION
+function set_binary_version() {
+  if [[ "${1}" =~ "/" ]]; then
+    IFS='/' read -a path <<< "${1}"
+    if [[ "${path[0]}" == "release" ]]; then
+      KUBE_VERSION=$(curl "https://storage.googleapis.com/kubernetes-release/${1}.txt")
```

Comment:
even better: now we can just do `curl -L "https://dl.k8s.io/$1"`, which does the right thing for both `release/` and `ci/`:
https://dl.k8s.io/release/stable.txt
https://dl.k8s.io/ci/latest-1.6.txt